source/Octopus.Tentacle.Tests.Integration/CapabilitiesServiceV2Test.cs

```diff
@@ -51,5 +59,57 @@ public class CapabilitiesServiceV2Test : IntegrationTest
                 capabilities.Count.Should().Be(2);
             }
         }
+
+        [Test]
+        [TestCaseSource(typeof(CapabilitiesServiceInterestingTentacles))]
+        public async Task CapabilitiesResponseShouldBeCached(TentacleType tentacleType, string? version)
+        {
+            var capabilitiesResponses = new List<CapabilitiesResponseV2>();
+            var resumePortForwarder = false;
+
+            using var clientAndTentacle = await new ClientAndTentacleBuilder(tentacleType)
+                .WithPortForwarder(out var portForwarder)
+                .WithTentacleVersion(version)
+                .WithTentacleServiceDecorator(new TentacleServiceDecoratorBuilder()
+                    .DecorateCapabilitiesServiceV2With(d => d
+                        .AfterGetCapabilities((response) =>
+                        {
+                            capabilitiesResponses.Add(response);
+
+                            if (resumePortForwarder)
```

Comment:
Oh this port forwarder is needed since the decorators wrap halibut and it is halibut doing the caching so unless we do this we wont know if it is coming from the cache or by a call to tentacle.

I guess this is where some sort of tentacle proxy would be super useful.